packages/router/router.ts

```diff
@@ -616,18 +621,14 @@ interface ShortCircuitable {
   shortCircuited?: boolean;
 }
 
+type PendingActionResult = [string, SuccessResult | ErrorResult];
+
 interface HandleActionResult extends ShortCircuitable {
   /**
-   * Error thrown from the current action, keyed by the route containing the
-   * error boundary to render the error.  To be committed to the state after
-   * loaders have completed
-   */
-  pendingActionError?: RouteData;
-  /**
-   * Data returned from the current action, keyed by the route owning the action.
-   * To be committed to the state after loaders have completed
+   * Tuple for the returned or thrown value from the current action.  The routeId
+   * is the action route for success and the bubbled boundary route for errors.
    */
-  pendingActionData?: RouteData;
+  pendingActionResult?: PendingActionResult;
```

Comment:
Combined these into a single data structure that allows us to carry along the routeId and the full `SuccessResult`/`ErrorResult` which contain the `statusCode` needed to implement `unstable_skipActionErrorRevalidation` behavior.  Previously we only had the routeId and the data in `RouteData`